core/services/ocrcommon/data_source.go

```diff
@@ -270,7 +296,13 @@ func (ds *inMemoryDataSourceCache) get(ctx context.Context) (pipeline.FinalResul
 }
 
 func (ds *inMemoryDataSourceCache) Observe(ctx context.Context, timestamp ocr2types.ReportTimestamp) (*big.Int, error) {
+	var val serializablebig.Big
 	latestResult, latestTrrs := ds.get(ctx)
+	if latestTrrs == nil {
+		ds.lggr.Errorf("cache is empty, returning persisted value now")
```

Comment:
Would we not want to persist this value to the cache for future retrievals?

Reply:
Using the persisted value from KV Store should be the worst case scenario, so I would rather keep throwing errors if value is retrieved from there. The only benefit I can see is if NOP db also fails and the value from KV store can't be retrieved, but if that happens I don't think anything else would work either. Also it can't really elegantly replace cache values since those aren't just big ints but whole task runs which I can't store in the KV store